src/Features/Core/Portable/EditAndContinue/Remote/RemoteEditAndContinueServiceProxy.cs

```diff
@@ -0,0 +1,462 @@
+                return await GetLocalService().IsActiveStatementInExceptionRegionAsync(instructionId, cancellationToken).ConfigureAwait(false);
+            }
+
+            return await client.RunRemoteAsync<bool?>(
+                WellKnownServiceHubService.RemoteEditAndContinueService,
+                nameof(IRemoteEditAndContinueService.IsActiveStatementInExceptionRegionAsync),
+                solution: null,
+                new object[] { moduleId, methodToken, methodVersion, ilOffset },
+                callbackTarget: null,
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task<ImmutableArray<ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>>> GetBaseActiveStatementSpansAsync(ImmutableArray<DocumentId> documentIds, CancellationToken cancellationToken)
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
+            if (client == null)
+            {
+                return await GetLocalService().GetBaseActiveStatementSpansAsync(documentIds, cancellationToken).ConfigureAwait(false);
+            }
+
+            var result = await client.RunRemoteAsync<ImmutableArray<ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>>>(
+                WellKnownServiceHubService.RemoteEditAndContinueService,
+                nameof(IRemoteEditAndContinueService.GetBaseActiveStatementSpansAsync),
+                solution: null,
+                new object[] { documentIds },
+                callbackTarget: null,
+                cancellationToken).ConfigureAwait(false);
+
+            return result;
+        }
+
+        public async Task<ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>> GetAdjustedActiveStatementSpansAsync(Document document, DocumentActiveStatementSpanProvider activeStatementSpanProvider, CancellationToken cancellationToken)
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
+            if (client == null)
+            {
+                return await GetLocalService().GetAdjustedActiveStatementSpansAsync(document, activeStatementSpanProvider, cancellationToken).ConfigureAwait(false);
+            }
+
+            var result = await client.RunRemoteAsync<ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>?>(
+                WellKnownServiceHubService.RemoteEditAndContinueService,
+                nameof(IRemoteEditAndContinueService.GetAdjustedActiveStatementSpansAsync),
+                document.Project.Solution,
+                new object[] { document.Id },
+                callbackTarget: new DocumentActiveStatementProviderCallback(activeStatementSpanProvider),
+                cancellationToken).ConfigureAwait(false);
+
+            // JSON-RPC does not support serialization of default ImmutableArray
+            return result ?? default;
```

Comment:
oh, that's cleer.  i ran into this myself, and solved it just passing arrays.  i far prefer your solution.